File: FbxOpenCpp/AnimStackTest.cpp

#include "Tests.h"

using namespace std;

void FbxAnimStack_Create_HasNamespacePrefix()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxAnimStack* obj = FbxAnimStack::Create(manager, "asdf");

    // then:
    AssertEqual("AnimStack::", obj->GetNameSpacePrefix());
}

void AnimStackTest::RegisterTestCases()
{
    AddTestCase(FbxAnimStack_Create_HasNamespacePrefix);
}

File: FbxSharpTests/AnimStackTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class AnimStackTest : TestBase
    {
        [Test]
        public void FbxAnimStack_Create_HasNamespacePrefix()
        {
            // given:
            var obj = new FbxAnimStack("asdf");

            // then:
            Assert.AreEqual("AnimStack::", obj.GetNameSpacePrefix());
        }
    }
}
